### PR TITLE
feat(pkg/db): remove cleveldb, fsdb and rocksdb

### PR DESCRIPTION
**Pull request by thehowl**
Split from #1602. Context:

> @zivkovicmilos:[^1] Unpopular opinion: we should probably just drop these DBs entirely from the codebase, they are never gonna be used in place of LevelDB in the near future
>
> @ajnavarro: IMHO the needed database is defined by the application needs, not from any other external usecases. Sticking to one of the existing key/value databases, we can use their special capabilities to make it more performant for our use case (yes, even if they are all key/value DBs, they differ in a lot of low-level functionality). \
> I would say to use one and add support in the future for more if needed.

After a discussion in the review meeting w/  @jaekwon, this PR is now changed to maintain another reference implementation with boltdb. It also makes sure that it is selectable, when available, from the tm2 node configuration.

After this PR, there are no databases in the codebase using cgo.

[^1]: https://github.com/gnolang/gno/pull/1602#discussion_r1490862277

### CODE REVIEW

File: tm2/pkg/db/goleveldb/go_level_db_test.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/gnolang/gno/tm2/pkg/db/internal"
 	"github.com/gnolang/goleveldb/leveldb/opt"
 	"github.com/jaekwon/testify/assert"
-	"github.com/stretchr/testify/require"
+	"github.com/jaekwon/testify/require"
```
Comment (ajnavarro) [inline tm2/pkg/db/goleveldb/go_level_db_test.go:11]: Do  we need to modify this on this PR?

**Review by ajnavarro — APPROVED** [review]

File: tm2/pkg/db/goleveldb/go_level_db_test.go (hunk 1 above)
Comment (thehowl) [reply tm2/pkg/db/goleveldb/go_level_db_test.go:11]: Here I was making it consistent with the above import, not crucial but I think it doesn't hurt...

Comment (zivkovicmilos) [reply tm2/pkg/db/goleveldb/go_level_db_test.go:11]: We should have a discussion, outside this PR, on these packages

**Review by zivkovicmilos — APPROVED** [review]
3nkh17

File: tm2/pkg/db/goleveldb/go_level_db_test.go (hunk 1 above)
Comment (harry-hov) [reply tm2/pkg/db/goleveldb/go_level_db_test.go:11]: afaik, we already had a discussion about this and we settled on using the jaekwon's one.

Ref: https://github.com/gnolang/gno/issues/777#issuecomment-1632014195

**Comment by thehowl** [comment]
From the review meeting: let's keep one additional implementation aside from the ones that are currently used.

For what @ajnavarro was referring to, we can support functionality of specific databases by adding extra methods, and then using type assertions in usage to verify if they support that (this would likely have to be done anyway, even if we keep as it is now only goleveldb and memdb.)

I will update the PR and add support boltdb

**Comment by thehowl** [comment]
This PR is now updated, and ready for a last review before merging.

**Pull request merged**